### PR TITLE
Rename http_client to http_proxy and refactor code

### PR DESCRIPTION
**Pull request by ahsank**

### CODE REVIEW

File: src/k2/httpproxy/http_proxy.h [hunk 1]
```diff
@@ -21,18 +21,21 @@ Copyright(c) 2021 Futurewei Cloud
     SOFTWARE.
```
Comment (iafuture) [inline src/k2/httpproxy/http_proxy.h:21]: - You should name the files with UpperCamelCase.
- The header file should be split into header and implementation.  All implementation should move from the header to  the .cpp file.
- this folder needs a CMakeLists.txt file
- this folder needs to be added to the parent CMakeLists.txt file

**Review by iafuture — CHANGES_REQUESTED** [review]

File: src/k2/httpproxy/http_proxy.h (hunk 1 above)
Comment (ahsank) [reply src/k2/httpproxy/http_proxy.h]: Pushed a new commit where I renamed `http_proxy.h` to `HTTPProxy.h` and moved implementation to `HTTPProxy.cpp` and created library `libhttproxy`. Please review again.

**Review by jfunston — DISMISSED** [review]

File: src/k2/cmd/httpproxy/http_proxy_main.cpp [hunk 2]
```diff
@@ -20,10 +20,24 @@ Copyright(c) 2021 Futurewei Cloud
...
+#include <k2/appbase/AppEssentials.h>
+#include <k2/appbase/Appbase.h>
+#include <k2/infrastructure/APIServer.h>
+#include <k2/tso/client/Client.h>
+#include <k2/httpproxy/HTTPProxy.h>
 
-namespace k2::log {
-inline thread_local k2::logging::Logger httpclient("k2::httpclient");
+int main(int argc, char** argv) {
+    k2::App app("K23SIBenchProxy");
```
Comment (iafuture) [inline src/k2/cmd/httpproxy/http_proxy_main.cpp:31]: Please rename to HttpProxy

File: src/k2/cmd/httpproxy/http_proxy_main.cpp [hunk 3]
```diff
@@ -20,10 +20,24 @@ Copyright(c) 2021 Futurewei Cloud
...
+    app.addApplet<k2::APIServer>();
+    app.addApplet<k2::tso::TSOClient>();
+    app.addApplet<k2::HTTPProxy>();
+    app.addOptions()
+        // config for dependencies
+        ("num_partitions", bpo::value<uint32_t>(), "Number of k2 nodes to use for the collection")
```
Comment (iafuture) [inline src/k2/cmd/httpproxy/http_proxy_main.cpp:37]: Should make an issue for later to also configure the HTTP port for the proxy

File: src/k2/httpproxy/HTTPProxy.cpp [hunk 4]
```diff
@@ -0,0 +1,398 @@
+#include "HTTPProxy.h"
```
Comment (iafuture) [inline src/k2/httpproxy/HTTPProxy.cpp:1]: Please add copyright header for 2022

File: src/k2/httpproxy/HTTPProxy.h [hunk 5]
```diff
@@ -0,0 +1,71 @@
+/*
+MIT License
+
+Copyright(c) 2021 Futurewei Cloud
```
Comment (iafuture) [inline src/k2/httpproxy/HTTPProxy.h:4]: Please change year to 2022

**Review by iafuture — CHANGES_REQUESTED** [review]

File: src/k2/cmd/httpproxy/http_proxy_main.cpp (hunk 2 above)
Comment (ahsank) [reply src/k2/cmd/httpproxy/http_proxy_main.cpp]: Fixed

File: src/k2/httpproxy/HTTPProxy.cpp (hunk 4 above)
Comment (ahsank) [reply src/k2/httpproxy/HTTPProxy.cpp:24]: Fixed

File: src/k2/httpproxy/HTTPProxy.h (hunk 5 above)
Comment (ahsank) [reply src/k2/httpproxy/HTTPProxy.h]: Fixed

**Review by iafuture — APPROVED** [review]

**Pull request merged**